xapian-applications/omega/omega.cc

```diff
@@ -457,6 +494,10 @@ try {
     if (val != cgi_params.end()) {
 	const string & v = val->second;
 	if (!v.empty()) {
+	    if (v[0] < '0' || v[0] > '9' || v[0] == '-') {
+		cerr << "Value entered is not in range\n"
+		    << "Range is a positive number";
```

Comment:
`0` is valid here, but "positive" suggests otherwise.  "Non-negative" would be clearer.

But all these messages also suffer from not really giving enough context - something like "COLLAPSE parameter must be >= 0" would be clearer still.  Otherwise when multiple parameters are passed you'd have to hunt through them to work out which is actually the problem.

Reply:
thanks i will fix that.